### PR TITLE
feat: implementation of short term material demand in the dashboard

### LINKED ISSUE #205: Dashboard Integration for the needed filters for Short-Term Material Demand

### PR DESCRIPTION
**Pull request by ReneSchroederLJ**
## Description

- implemented short term material demand in the frontend

resolves #205 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: frontend/src/features/dashboard/hooks/useReportedDemand.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { useFetch } from '@hooks/useFetch';
+import { config } from '@models/constants/config';
+import { Demand } from '@models/types/data/demand';
+
+export const useReportedDemand = (materialNumber: string | null) => {
+  const {data: reportedDemands, error: reportedDemandsError, isLoading: isLoadingReportedDemands, refresh: refreshDemand } = useFetch<Demand[]>(materialNumber ? `${config.app.BACKEND_BASE_URL}${config.app.ENDPOINT_DEMAND}/reported?materialNumber=${materialNumber}` : undefined);
```
[the remark this answers is not in the document]
Comment (ReneSchroederLJ) [reply frontend/src/features/dashboard/hooks/useReportedDemand.ts:6]: Very likely yes. If time permits I would like to unify the standard endpoints so they all use the same structure. Then we wouldn't need more than the base url anymore.

File: frontend/src/features/dashboard/components/DemandCategoryModal.tsx [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+                                            error={formError && !temporaryDemand?.demandCategoryCode}
+                                        />
+                                    )}
+                                />
+                            </Grid>
+                            <Grid item xs={6}>
+                                <Input
+                                    label="Quantity*"
+                                    type="number"
+                                    placeholder='Enter quantity'
```
[the remark this answers is not in the document]
Comment (ReneSchroederLJ) [reply frontend/src/features/dashboard/components/DemandCategoryModal.tsx:255]: No. This is an oversight and should be corrected when possible. The formatter should catch this normally, but doesn't seem to do it always.

File: frontend/src/features/dashboard/components/Dashboard.tsx [hunk 3]
```diff
@@ -90,9 +102,16 @@
...
+            dispatch({ type: 'delivery', payload: d });
+            dispatch({ type: 'deliveryDialogOptions', payload: { open: true, mode: 'edit' } });
+        };
+    const openDemandDialog = (d: Partial<Demand>, mode: 'create' | 'edit') => {
+        d.measurementUnit ??= 'unit:piece';
+        d.demandCategoryCode ??= DEMAND_CATEGORY[0]?.key;
+        d.ownMaterialNumber = state.selectedMaterial?.ownMaterialNumber ?? '';
+        dispatch({ type: 'demand', payload: d });
+        dispatch({ type: 'demandDialogOptions', payload: { open: true, mode } });
+    }
```
[the remark this answers is not in the document]
Comment (tom-rm-meyer-ISST) [reply frontend/src/features/dashboard/components/Dashboard.tsx:114]: @ReneSchroederLJ please update

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
LGTM, please incorporate the CodeQl finding and I'll merge :)

**Review by tom-rm-meyer-ISST — APPROVED** [review]
LGTM, thanks!

**Pull request merged**